addons/portal/wizard/portal_wizard.py

```diff
@@ -234,24 +212,8 @@ def _send_email(self, cr, uid, wizard_user, context=None):
                                                             context=ctx_portal_url)[user.partner_id.id]
         res_partner.signup_prepare(cr, uid, [user.partner_id.id], context=context)
 
-        data = {
-            'company': this_user.company_id.name,
-            'portal': wizard_user.wizard_id.portal_id.name,
-            'welcome_message': wizard_user.wizard_id.welcome_message or "",
-            'db': cr.dbname,
-            'name': user.name,
-            'login': user.login,
-            'signup_url': user.signup_url,
-            'portal_url': portal_url,
-        }
-        mail_mail = self.pool.get('mail.mail')
-        mail_values = {
-            'email_from': this_user.email,
-            'email_to': user.email,
-            'subject': _(WELCOME_EMAIL_SUBJECT) % data,
-            'body_html': '<pre>%s</pre>' % (_(WELCOME_EMAIL_BODY) % data),
-            'state': 'outgoing',
-            'type': 'email',
-        }
-        mail_id = mail_mail.create(cr, uid, mail_values, context=this_context)
-        return mail_mail.send(cr, uid, [mail_id], context=this_context)
+        context.update({'dbname': cr.dbname, 'portal_url': portal_url, 'user': user})
```

Comment:
I think we could improve the way users are managed here. I would add a field on portal.wizard.user, user_id, many2one towards a res.users field. When you do a _retrieve_users, update the portal_user.user_id field, so in the template and everywhere else in the code you can use the user_id field instead of having to retrieve it every time and use it in the context. Could you update the code please ? Thanks !